models/attribution.test.js

```diff
@@ -0,0 +1,316 @@
+
+describe('attribution', () => {
+  const exampleCC4License = new License({
+    id: 'cc-by-sa-4.0',
+    name: 'CC BY-SA 4.0',
+    groups: ['cc', 'cc4'],
+    compatibility: [],
+    regexp: /^(Bild-)?CC-BY-SA(-|\/)4.0(([^-]+.+|-migrated)*)?$/i,
+    url: 'https://creativecommons.org/licenses/by-sa/4.0/legalcode',
+  });
+
+  const exampleCC2License = new License({
+    id: 'cc-by-sa-2.5',
+    name: 'CC BY-SA 2.5',
+    groups: ['cc', 'cc2.5', 'ccby'],
+    compatibility: [],
+    regexp: /^CC-BY-2.5-\w+$/i,
+    url: 'https://creativecommons.org/licenses/by-sa/2.5/legalcode',
+  });
+
+  const examplePublicDomainLicense = new License({
+    id: 'PD-1923',
+    name: 'Public Domain',
+    groups: ['pd'],
+    compatibility: ['cc-by-4.0', 'cc-by-sa-4.0'],
+    regexp: /PD-1923/,
+    url: 'https://commons.wikimedia.org/wiki/Template:PD-1923',
+  });
+
+  const options = {
+    fileInfo: {
+      rawUrl: 'https://commons.wikimedia.org/wiki/File:Eisklettern_kl_engstligenfall.jpg',
+      title: 'File:Eisklettern kl engstligenfall.jpg',
+      artistHtml:
+        '<a href="//commons.wikimedia.org/w/index.php?title=User:Bernhard&amp;action=edit&amp;redlink=1" class="new" title="User:Bernhard (page does not exist)">Bernhard</a>',
+      attributionHtml: null,
+    },
+    typeOfUse: 'online',
+    languageCode: 'de',
+    license: exampleCC2License,
+    modification: null,
+    modificationAuthor: null,
+    isEdited: false,
+  };
+
+  function newAttribution(overrides = {}) {
+    return new Attribution({ ...options, ...overrides });
+  }
+
+  it('initalizes', () => {
+    newAttribution();
+  });
+
+  describe('validations', () => {
+    it('asserts valid fileInfo.rawUrl', () => {
+      expect(() => newAttribution({ fileInfo: { rawUrl: 123, title: 'title' } })).toThrow();
+    });
+
+    it('asserts valid fileInfo.title', () => {
+      expect(() => newAttribution({ fileInfo: { rawUrl: 'url', title: '' } })).toThrow();
+    });
+
+    it('asserts valid typeOfUse', () => {
+      expect(() => newAttribution({ typeOfUse: 'print' })).toThrow();
+    });
+
+    it('asserts valid languageCode', () => {
+      expect(() => newAttribution({ languageCode: 'yo' })).toThrow();
+    });
+
+    it('asserts valid fileInfo.artistHtml', () => {
+      expect(() => newAttribution({ fileInfo: { artistHtml: 123 } })).toThrow();
+    });
+
+    it('asserts valid fileInfo.attributionHtml', () => {
+      expect(() => newAttribution({ fileInfo: { attributionHtml: 123 } })).toThrow();
+    });
+
+    it('asserts valid license', () => {
+      expect(() => newAttribution({ license: 'public domain' })).toThrow();
+    });
+
+    it('asserts valid modification', () => {
+      expect(() => newAttribution({ modification: 123 })).toThrow();
+    });
+
+    it('asserts valid modificationAuthor', () => {
+      expect(() => newAttribution({ modificationAuthor: 123 })).toThrow();
+    });
+  });
+
+  describe('when initialized with defaults', () => {
```

Comment:
How about doing all of these as snapshots? Does not clutter the file that much...